deep_agent/aegra/mcp_tool_auth.py

```
@@ -80,6 +80,39 @@ def wrap_mcp_tools_for_auth(tools: list[Any]) -> list[Any]:
+        from langchain_core.messages import ToolMessage
+
+        try:
+            return await original_ainvoke(tool_input, config, **kwargs)
+        except NeedsAuthorization as exc:
+            logger.info(
+                "MCP auth required for '%s' — interrupting run",
+                exc.mcp_name,
+            )
+            interrupt(_mcp_auth_interrupt_payload(exc))
+            return await original_ainvoke(tool_input, config, **kwargs)
+        except Exception as exc:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)